src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1317,18 +1317,17 @@ private boolean confirmClose(BasePanel panel) {
             // The user wants to save.
             try {
                 SaveDatabaseAction saveAction = new SaveDatabaseAction(panel, Globals.prefs);
-                if (!saveAction.save()) {
-                    // The action was either canceled or unsuccessful.
-                    output(Localization.lang("Unable to save library"));
-                    return false;
+                if (saveAction.save()) {
+                    // Saved, now exit.
+                    return true;
                 }
+                // The action was either canceled or unsuccessful.
+                output(Localization.lang("Unable to save library"));
             } catch (Throwable ex) {
-                return false;
+                LOGGER.error("A problem occurred when saving the file", ex);
             }
-        } else {
-            return !response.isPresent() || !response.get().equals(cancel);
         }
-        return false;
+        return !response.isPresent() || !response.get().equals(cancel);
```

Comment:
Here it is sufficient to return simply false, because you do the check already as the very first. And the user can only click save or cancel.

Reply:
Aren't there 3 options, _Save changes_, _Discard changes_ and _Return to JabRef_?

Reply:
If only return false is used, then _Discard changes_ will not quit the application, is that the intended behavior?

Reply:
You are right. Forgot that there is a third option